debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/SourceInfo.java

```diff
@@ -296,6 +296,10 @@ public boolean setOffsetFor(String replicaSetName, Map<String, ?> sourceOffset)
         if (replicaSetName == null) throw new IllegalArgumentException("The replica set name may not be null");
         if (sourceOffset == null) return false;
         // We have previously recorded at least one offset for this database ...
+        boolean initSync = booleanOffsetValue(sourceOffset, INITIAL_SYNC);
+        if (initSync) {
```

Comment:
This does not seem to work as expected for me.

AFAICS, the replica set name would have to be added to `initialSyncReplicaSets` in this case, so that `SourceInfo#isInitialSyncOngoing()` returns true, resulting in the snapshot being taken again. WDYT?

Reply:
@gunnarmorling I don't quite understand your question, but the issue I am fixing is that when the initial sync is interrupted and when the connector is started again, it doesn't redo the initial sync, and the reason is in this code, it ignores the `initsync` flag and put the offset in `positionsByReplicaSetName`, then in `Replicator.isInitialSyncExpected()` it returns false.

I tested this fix in my local env and it works well.

Reply:
@echo-xu What I meant is that your fix doesn't seem to solve the issue for me.

I.e. unless I missed something, it's still not taking a new snapshot if the first one is interrupted. The reason being, that the code doesn't answer `if (source.isInitialSyncOngoing(rsName))` with `true` in [Replicator, line 205](https://github.com/debezium/debezium/blob/master/debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java#L205).

> doesn't return true in this case because the the replica set name is only added to initialSyncReplicaSets in startInitialSync method.

Right, hence my suggestion to add the replica set name to `initialSyncReplicaSets`.

Reply:
@gunnarmorling with this fix, it will go to https://github.com/debezium/debezium/blob/master/debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/Replicator.java#L236 and then perform a snapshot.

Reply:
Yeah, I can confirm that now. Thanks, going to merge this PR.